control-plane/api/v1alpha1/peeringdialer_types.go

```diff
@@ -0,0 +1,53 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+func init() {
+	SchemeBuilder.Register(&PeeringDialer{}, &PeeringDialerList{})
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// PeeringDialer is the Schema for the peeringdialers API.
+type PeeringDialer struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PeeringDialerSpec   `json:"spec,omitempty"`
+	Status PeeringDialerStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// PeeringDialerList contains a list of PeeringDialer.
+type PeeringDialerList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []PeeringDialer `json:"items"`
+}
+
+// PeeringDialerSpec defines the desired state of PeeringDialer.
+type PeeringDialerSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Peer describes the information needed to create a peering.
+	Peer *Peer `json:"peer"`
+}
+
+// PeeringDialerStatus defines the observed state of PeeringDialer
+type PeeringDialerStatus struct {
```

Comment:
This is a TODO for later, but we should introduce Conditions here which are a good way to express errors and if things worked. Statuses like `ConditionReady` etc can come in handy.

Reply:
Added a cleanup TODO to the Asana!